src/simd_fnt.h

```diff
@@ -37,42 +37,55 @@ namespace quadiron {
 namespace simd {
 
 template <typename T>
-inline VecType card(T q);
+inline VecType card();
 template <>
-inline VecType card<uint16_t>(uint16_t)
+inline VecType card<uint16_t>()
 {
-    return F3_U16;
+    return set_one<uint16_t>(257);
 }
 template <>
-inline VecType card<uint32_t>(uint32_t q)
+inline VecType card<uint32_t>()
 {
-    return (q == F3) ? F3_U32 : F4_U32;
+    return set_one<uint32_t>(65537);
 }
 
 template <typename T>
-inline VecType card_minus_one(T q);
+inline VecType card_minus_one();
 template <>
-inline VecType card_minus_one<uint16_t>(uint16_t)
+inline VecType card_minus_one<uint16_t>()
 {
-    return F3_MINUS_ONE_U16;
+    return set_one<uint16_t>(256);
 }
 template <>
-inline VecType card_minus_one<uint32_t>(uint32_t q)
+inline VecType card_minus_one<uint32_t>()
 {
-    return (q == F3) ? F3_MINUS_ONE_U32 : F4_MINUS_ONE_U32;
+    return set_one<uint32_t>(65536);
 }
 
 template <typename T>
-inline VecType get_low_half(VecType x, T q)
+inline VecType get_low_half(const VecType& x);
+template <>
+inline VecType get_low_half<uint16_t>(const VecType& x)
+{
+    return BLEND8(zero(), x, MASK8_LO);
+}
+template <>
+inline VecType get_low_half<uint32_t>(const VecType& x)
 {
-    return (q == F3) ? BLEND8(ZERO, x, MASK8_LO) : BLEND16(ZERO, x, 0x55);
+    return BLEND16(zero(), x, 0x55);
```

Comment:
Could we have a document constant (like we did with `MASK8_LO`) instead of having a raw magic number?

Reply:
okay, it's better